docs/config/environment/advanced.md

```diff
@@ -127,6 +127,71 @@ you could then run your tests consecutively in all 4 environments with:
 hatch run test:cov
 ```
 
+## Dependency Groups
+
+Environments can use [PEP 735](https://peps.python.org/pep-0735/) dependency groups with the `dependency-groups` option:
+
+```toml config-example
+[dependency-groups]
+test = [
+  "pytest>=7.0.0",
+  "pytest-cov>=4.1.0",
+]
+lint = [
+  "black",
+  "ruff",
+  "mypy",
+]
+# Groups can include other groups
+dev = [
+  {"include-group": "test"},
+  {"include-group": "lint"},
+  "pre-commit",
+]
+
+[tool.hatch.envs.test]
+dependency-groups = ["test"]
+
+[tool.hatch.envs.lint]
+dependency-groups = ["lint"]
+
+[tool.hatch.envs.dev]
+dependency-groups = ["dev"]
```

Comment:
i can imagine that a completely reflected use-case would be preferable here. i don't know the current affairs reg. user documentation guidelines, maybe @lwasser has some advice.